staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go

```diff
@@ -61,6 +61,8 @@ const (
 // Config contains the configuration for a given Cache.
 type Config struct {
 	// Maximum size of the history cached in memory.
+	//
+	// DEPRECATED: Cache capacity is dynamic and this field is no longer used.
```

Comment:
Can we not keep this as a *maximum*? I thought we had agreed that a cap was useful?

Reply:
There is a cap. With this change it will simply always default to this value - https://github.com/kubernetes/kubernetes/blob/442a69c3bdf6fe8e525b05887e57d89db1e2f3a5/staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache.go#L61

Reply:
Yes - the cap is for sure needed. And it already there as Matt wrote above.

I also discussed that with @jpbetz some time ago and the conclusion was that different caps for different resources is probably an overkill.

Reply:
OK if you don't need different caps for different types, then this seems fine.

Reply:
Yeah - the cap is more like a last resort and shouldn't really be reached. So introducing more complication about that seems unneeded (at least for now where we don't have any evidence for the need of it).